modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -1543,4 +1550,29 @@ private void saveRaftGroupConfigurationOnRebalance(WriteBatch writeBatch, RaftGr
 
         this.lastGroupConfig = config;
     }
+
+    /**
+     * Prepares the storage for cleanup.
+     *
+     * <p>After cleanup (successful or not), method {@link #finishCleanup()} must be called.
+     */
+    void startCleanup(WriteBatch writeBatch) throws RocksDBException {
+        if (!state.compareAndSet(StorageState.RUNNABLE, StorageState.CLEANUP)) {
+            throwExceptionDependingOnStorageState(state.get(), createStorageInfo());
+        }
+
+        // Changed storage states and expect all storage operations to stop soon.
+        busyLock.block();
+
+        clearStorage(writeBatch, 0, 0);
+    }
+
+    /**
+     * Finishes cleanup up the storage.
+     */
+    void finishCleanup() {
+        if (state.compareAndSet(StorageState.CLEANUP, StorageState.RUNNABLE)) {
+            busyLock.unblock();
```

Comment:
I wonder why in AbstractPageMemoryMvPartitionStorage` `unblock()` is called in `startCleanup()`, but here it is called in `finishCleanup()`?

Reply:
Here we execute in same thread and in `AbstractPageMemoryMvPartitionStorage` in different ones, so the implementation is slightly different.

Reply:
I see